### PR TITLE
Session migration

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

File: src/room/track/TrackPublication.ts [hunk 1]
```diff
@@ -27,6 +27,8 @@ export class TrackPublication extends EventEmitter {
   /** true if track was simulcasted to server, video-only */
   simulcasted?: boolean;
 
+  trackInfo?: TrackInfo;
```
Comment (davidzhao) [inline src/room/track/TrackPublication.ts:30]: also would be great to mark as @internal

**Review by davidzhao — COMMENTED** [review]
Looks good! The main question I had was `sendUnsub` usage.

File: src/room/Room.ts [hunk 2]
```diff
@@ -687,6 +699,36 @@ class Room extends EventEmitter {
     return participant;
   }
 
+  private sendSyncState() {
+    if (this.engine.subscriber === undefined
+      || this.engine.subscriber.pc.localDescription === null) {
+      return;
+    }
+    const previousSdp = this.engine.subscriber.pc.localDescription;
+    const sendUnsub = this.connOptions?.autoSubscribe || false;
```
[the remark this answers is not in the document]
Comment (cnderrauber) [reply src/room/Room.ts:708]: I think there are two ways:
1. autosubscribe on, so subscribed tracks = all tracks - unsub tracks, in this case, we send unsub tracks, so server add all tracks to this subscribe pc and unsub special tracks from it.
2. autosubscribe off, we send subscribed tracks.

[the remark this answers is not in the document]
Comment (davidzhao) [reply src/room/Room.ts:708]: that makes sense. thank you for the explanation. So the server would have to know to treat this as an exclusion-list if it sees [subscribe: false] in the tracks.

**Review by davidzhao — APPROVED** [review]

File: src/room/Room.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cnderrauber) [reply src/room/Room.ts:708]: will do

[the remark this answers is not in the document]
Comment (boks1971) [reply src/room/Room.ts:708]: Extending this discussion a bit, sorry.

Even in the case of `autoSubscribe = false`, a participant could have subscribed to some tracks and subsequently unsubscribed. They would have consumed space in the SDP (if those transceivers were not re-used).

Can we send them all? Can we not use SDP answer (maybe send last SDP offer of subscriber too if necessary) to reconcile tracks/transceivers and their states in the SDP?

**Pull request merged**